common/setup.py

```diff
@@ -28,14 +28,16 @@
         "tqdm",  # version range defined in `core/_setup_utils.py`
         # s3fs is removed due to doubling install time due to version range resolution
         # "s3fs",  # version range defined in `core/_setup_utils.py`
-        "setuptools",
+        "setuptools<60.0; python_version < '3.12'",
+        "setuptools>=60.0; python_version >= '3.12'",
```

Comment:
Do we need setuptools?

Reply:
Upon testing, I think it can be removed.